src/factories.js

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import components from './index';
+
+let factories = {};
+
+Object.keys(components).forEach(function (name) {
+  if (components[name] instanceof React.Component.constructor) {
```

Comment:
Good point. I'm working on test cases and will be sure to check. However, there is a minor downside to this approach. Since there are no actual files, I cannot import like this:

    var Alert = require('react-bootstrap/lib/factories/alert')

I'm considering alternatives to make a recommendation. Do you have an opinion on adding files for the factories? It may be possible to generate them automatically in the build process...

    var Alert = require('react-bootstrap/lib/alert-factory')